src/ResourceManager/DataFactories/Commands.DataFactories/ActivityWindows/GetAzureDataFactoryActivityWindowsCommand.cs

```diff
@@ -0,0 +1,132 @@
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using Microsoft.Azure.Commands.DataFactories.Models;
+using Microsoft.Azure.Commands.DataFactories.Properties;
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Management.Automation;
+using System.Security.Permissions;
+
+namespace Microsoft.Azure.Commands.DataFactories
+{
+    [Cmdlet(VerbsCommon.Get, Constants.ActivityWindows, DefaultParameterSetName = ByFactoryName), OutputType(typeof(List<PSActivityWindow>))]
+    public class GetAzureDataFactoryActivityWindowsCommand : DataFactoryBaseCmdlet
+    {
+        [Parameter(ParameterSetName = ByFactoryObject, Position = 0, Mandatory = true, ValueFromPipelineByPropertyName = true,
+            HelpMessage = "The data factory object.")]
+        public PSDataFactory DataFactory { get; set; }
+
+        [Parameter(ParameterSetName = ByFactoryName, Position = 1, Mandatory = true, ValueFromPipelineByPropertyName = true,
+            HelpMessage = "The data factory name.")]
+        [ValidateNotNullOrEmpty]
+        public string DataFactoryName { get; set; }
+
+        [Parameter(Position = 2, Mandatory = false, ValueFromPipelineByPropertyName = true, HelpMessage = "The dataset name.")]
+        [ValidateNotNullOrEmpty]
+        public string DatasetName { get; set; }
+
+        [Parameter(Position = 2, Mandatory = false, ValueFromPipelineByPropertyName = true, HelpMessage = "The pipeline name.")]
+        [ValidateNotNullOrEmpty]
+        public string PipelineName { get; set; }
+
+        [Parameter(Position = 3, Mandatory = false, ValueFromPipelineByPropertyName = true, HelpMessage = "The pipeline activity name.")]
+        [ValidateNotNullOrEmpty]
+        public string ActivityName { get; set; }
+
+        [Parameter(Position = 4, Mandatory = false, ValueFromPipelineByPropertyName = true, HelpMessage = "The state of the activity window. The possible states includes: None, Waiting, InProgress, Ready, Failed, and Skipped")]
+        [ValidateNotNullOrEmpty]
+        public string WindowState { get; set; }
+
+        [Parameter(Position = 4, Mandatory = false, ValueFromPipelineByPropertyName = true, HelpMessage = "The substate of the activity window. The possible substates includes: Canceled, timedOut, Validating," +
+                                                                                                          " ScheduledTime, DatasetDependencies, ComputeResources, ConcurrencyLimit, ActivityResume, Retry, Validation, ValidationRetry.")]
+        [ValidateNotNullOrEmpty]
+        public string WindowSubstate { get; set; }
+
+        [Parameter(Position = 4, Mandatory = false, ValueFromPipelineByPropertyName = true, HelpMessage = "The activity window filter based on Azure Search filter grammar. For more information see https://msdn.microsoft.com/en-us/library/azure/dn798921.aspx")]
```

Comment:
How can you have so many parameters with Position 4?  Are there meant to be multiple parameter sets?